[ENH] Added Abagen and NiMARE support.

ReinderVosDeWael opened the pull request:
This PR implements abagen and nimare into BrainStat. To that end, it also impements a new `interpolation` submodule within the mesh module which allows for the interpolation of surface label data to a volume. 

Note that this new interpolation module requires that the package pyembree is installed. This package is only available through conda-forge and is itself dependent on embree. As such, we should probably include separate installation instructions for this in the documentation.

ReinderVosDeWael commented:
I'll add neurosynth in this same PR. I've returned this PR to a draft until that's done.

sheyma commented on brainstat/context/utils.py:
```diff
@@ -0,0 +1,179 @@
+from brainstat.mesh.interpolate import surface_to_volume
+
+
+def mutli_surface_to_volume(
```
maybe changing function name into "multi_surface_to_volume"

ReinderVosDeWael replied:
Change multi_surface_to_volume into multi_surface_to_volume?

sheyma replied:
nope, I meant changing  mu**tl**i_surface_to_volume into mu**lt**i_surface_to_volume

sheyma reviewed (commented):
great PR!
Python CI seems to fail bcs of minor flake8 issues. 
Once failures in Python CI fixed, feel free to merge this PR to master.

ReinderVosDeWael commented:
Changed. this back to a normal PR. The final list of changes is as follows:

- Added brainstat.mesh.interpolate surface to volume interpolations.
- Implemented abagen.
- Implemented NIMARE, although we may wish to change implementation details at a later date.
- Implemented loading of MPC gradients through enigmatoolbox - this is intended as temporary for the OHBM abstract until we write a better BigBrain module.
